app/scripts/modules/netflix/pipeline/stage/canary/canaryScores.component.ts

```diff
@@ -0,0 +1,104 @@
+    }
+  }
+
+  public onUpdate() {
+    this.invalid = !(this.successful && this.unhealthy);
+    if (this.onChange) {
+      this.onChange({
+        successfulScore: this.successful ? this.successful.toString() : '',
+        unhealthyScore: this.unhealthy ? this.unhealthy.toString() : ''
+      });
+    }
+  }
+}
+
+class CanaryScoreConfigComponent implements ng.IComponentOptions {
+
+  public bindings: any = {
+    unhealthyScore: '<',
+    successfulScore: '<',
+    onChange: '&'
+  };
+  public controller: any = CanaryScoreConfigComponentCtrl;
+  public template = `
+    <div ng-if="$ctrl.hasExpressions" class="form-group">
+      <div class="col-md-2 col-md-offset-1 sm-label-right">
+        <label>Canary Scores</label>
+      </div>
+      <div class="col-md-9 form-control-static">
+        Expressions are currently being used for canary scores.
+      </div>
+    </div>
+    <div class="canary-score" ng-if="!$ctrl.hasExpressions">
+      <div class="form-group">
+        <div class="col-md-2 col-md-offset-1 sm-label-right">
+          <label>Unhealthy Score</label>
+          <help-field key="pipeline.config.canary.unhealthyScore"></help-field>
+        </div>
+        <div class="col-md-2">
+          <input type="number"
+                 required
+                 min="0"
+                 max="{{$ctrl.successful - 1}}"
+                 ng-model="$ctrl.unhealthy"
+                 ng-change="$ctrl.onUpdate()"
+                 class="form-control input-sm"/>
+        </div>
+        <div class="col-md-2 col-md-offset-1 sm-label-right">
+          <label>Successful Score</label>
+          <help-field key="pipeline.config.canary.successfulScore"></help-field>
+        </div>
+        <div class="col-md-2">
+          <input type="number"
+                 required
+                 min="{{$ctrl.unhealthy + 1}}"
+                 max="100"
+                 ng-model="$ctrl.successful"
+                 ng-change="$ctrl.onUpdate()"
+                 class="form-control input-sm"/>
+        </div>
+      </div>
+      <div class="row">
+        <div class="col-md-offset-1 col-md-10">
+          <div class="progress">
+              <div class="progress-bar progress-bar-danger" style="width: {{$ctrl.invalid ? 0 : $ctrl.unhealthy}}%"></div>
+              <div class="progress-bar progress-bar-warning" style="width: {{$ctrl.invalid ? 0 : 100 - ($ctrl.unhealthy + (100 - $ctrl.successful))}}%"></div>
+              <div class="progress-bar progress-bar-success" style="width: {{$ctrl.invalid ? 0 : 100 - $ctrl.successful}}%"></div>
+              <div class="progress-bar progress-bar-warning" style="width: {{$ctrl.invalid ? 100 : 0}}%"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+}
+
+export const CANARY_SCORE_CONFIG_COMPONENT = 'spinnaker.netflix.canary.scores.component';
+module(CANARY_SCORE_CONFIG_COMPONENT, [])
+  .component('canaryScores', new CanaryScoreConfigComponent());
```

Comment:
For consistency, classes in here should probably be named `*Scores` instead of `*Score`